sympy/functions/elementary/tests/test_trigonometric.py

```diff
@@ -440,6 +441,10 @@ def test_cos_rewrite():
     assert cos(x).rewrite(sin) == sin(x + pi/2, evaluate=False)
     assert cos(x).rewrite(csc) == 1/csc(-x + pi/2, evaluate=False)
     assert cos(sin(x)).rewrite(Pow) == cos(sin(x))
+    assert cos(x).rewrite(besselj) == sqrt(2)*sqrt(pi)*sqrt(x)*besselj(-S.Half, x)/2
+    assert Piecewise((cos(x).rewrite(besselj), Ne(x, 0)), (1, True)) == \
+        Piecewise(((sqrt(2)*sqrt(pi)*sqrt(x)*besselj(-S.Half, x)/2), Ne(x, 0)), (1, True))
```

Comment:
The point here is that the rewrite should return a Piecewise.

Reply:
Piecewise can be done only on cos, sec, cot, tan ?

Reply:
```python
    def _eval_rewrite_as_besselj(self, arg, **kwargs):
        from sympy.functions.special.bessel import besselj
        return Piecewise(((sqrt(2)*sqrt(pi)*sqrt(arg)*besselj(-S.Half, arg)/2), Ne(arg, 0)), (1, True))
```
@oscarbenjamin Hey there! This function `cos(x).rewrite(besselj)` returns a Piecewise output. However, I believe that sin doesn't need a Piecewise function. I use cos/sin for cot. `(cos(arg).rewrite(besselj, **kwargs)/sin(arg).rewrite(besselj, **kwargs))` This results in the output referring to the wrong form(cos is Piecewise and sin is not). I'm confused about what  I do.